File: artifacts/AMMARS FRESH-mobile/lib/utils/format_price.dart
import 'package:intl/intl.dart';

import '../config/brand_config.dart';
import '../models/exchange_rates.dart';

const priceSeparator = ' · ';

String formatPriceDisplay(
  num priceSsp,
  num priceUsd, {
  String? unit,
  ExchangeRates rates = ExchangeRates.defaults,
}) {
  final usd = '\$${priceUsd.toStringAsFixed(2)} USD';
  final ssp = 'SSP ${NumberFormat('#,###').format(priceSsp)}';
  final usg = 'USG ${NumberFormat('#,###').format(rates.usdToUsgAmount(priceUsd))}';
  final base = '$usd$priceSeparator$ssp$priceSeparator$usg';
  return unit != null ? '$base/$unit' : base;
}

String formatProductPrice(
  String currency,
  num priceSsp,
  num priceUsd, {
  String? unit,
  ExchangeRates rates = ExchangeRates.defaults,
}) {
  final suffix = unit != null ? '/$unit' : '';
  switch (currency) {
    case 'USD':
      return '\$${priceUsd.toStringAsFixed(2)} USD$suffix';
    case 'SSP':
      return 'SSP ${NumberFormat('#,###').format(priceSsp)}$suffix';
    case 'USG':
      return 'USG ${NumberFormat('#,###').format(rates.usdToUsgAmount(priceUsd))}$suffix';
    default:
      return 'SSP ${NumberFormat('#,###').format(priceSsp)}$suffix';
  }
}

String formatOrderTotal(
  String currency,
  num totalSsp,
  num totalUsd, {
  ExchangeRates rates = ExchangeRates.defaults,
}) {
  switch (currency) {
    case 'USD':
      return '\$${totalUsd.toStringAsFixed(2)} USD';
    case 'SSP':
      return 'SSP ${NumberFormat('#,###').format(totalSsp)}';
    case 'USG':
      return 'USG ${NumberFormat('#,###').format(rates.usdToUsgAmount(totalUsd))}';
    default:
      return 'SSP ${NumberFormat('#,###').format(totalSsp)}';
  }
}

bool isSupportedCurrency(String value) => BrandConfig.currencies.contains(value);

File: artifacts/AMMARS FRESH/src/lib/format-price.ts
import { brand, type BrandCurrency } from "@/lib/brand";
import { type ExchangeRates, DEFAULT_EXCHANGE_RATES, usdToUsg } from "@/lib/exchange-rate";

export const PRICE_SEPARATOR = " · ";

type Priced = { priceSSP: number; priceUSD: number };

/** Public landing showcase — all currencies on one line */
export function formatPriceDisplay(
  item: Priced,
  rates: ExchangeRates = DEFAULT_EXCHANGE_RATES,
  unit?: string | null,
) {
  const usd = `$${item.priceUSD.toFixed(2)} USD`;
  const ssp = `SSP ${item.priceSSP.toLocaleString()}`;
  const usg = `USG ${usdToUsg(item.priceUSD, rates).toLocaleString()}`;
  const base = `${usd}${PRICE_SEPARATOR}${ssp}${PRICE_SEPARATOR}${usg}`;
  return unit ? `${base}/${unit}` : base;
}

export const priceDisplayClassName = "text-sm font-bold text-primary tabular-nums";

/** Single-currency price for retailer shop & cart */
export function formatProductPrice(
  currency: BrandCurrency,
  item: Priced,
  unit?: string | null,
  rates: ExchangeRates = DEFAULT_EXCHANGE_RATES,
) {
  const suffix = unit ? `/${unit}` : "";
  if (currency === "USD") return `$${item.priceUSD.toFixed(2)} USD${suffix}`;
  if (currency === "SSP") return `SSP ${item.priceSSP.toLocaleString()}${suffix}`;
  return `USG ${usdToUsg(item.priceUSD, rates).toLocaleString()}${suffix}`;
}

export function formatOrderTotal(
  currency: BrandCurrency,
  totalSSP = 0,
  totalUSD = 0,
  rates: ExchangeRates = DEFAULT_EXCHANGE_RATES,
) {
  if (currency === "USD") return `$${totalUSD.toFixed(2)} USD`;
  if (currency === "SSP") return `SSP ${totalSSP.toLocaleString()}`;
  return `USG ${usdToUsg(totalUSD, rates).toLocaleString()}`;
}

export function formatLineTotal(
  currency: BrandCurrency,
  item: Priced & { quantity: number; unit?: string | null },
  rates: ExchangeRates = DEFAULT_EXCHANGE_RATES,
) {
  return formatProductPrice(
    currency,
    { priceSSP: item.priceSSP * item.quantity, priceUSD: item.priceUSD * item.quantity },
    item.unit,
    rates,
  );
}

export const supportedCurrencies = brand.currencies;
